HaloCEVR/WeaponHandler.cpp

```diff
@@ -642,7 +643,15 @@ Matrix4 WeaponHandler::GetDominantHandTransform() const
 		}
 		*/
 
-		controllerTransform.lookAt(actualControllerPos + toOffHand, upVector);
+		float userInput = Game::instance.c_WeaponSmoothingAmount->Value();
+		float maxSmoothing = 15.0f;		//15 is already a bit ridiculous but just incase people need that much smoothing. 
+
+		//A value of 0 causes weapons to invert so have just above. 
+		float clampedValue = std::clamp(userInput, 0.0f, 0.9f);	
+		float smoothFactor = (1 - clampedValue) * maxSmoothing;
+
+		smoothedPosition = clampedValue == 0 ? actualControllerPos + toOffHand : Helpers::Lerp(smoothedPosition, actualControllerPos + toOffHand, smoothFactor * Game::instance.lastDeltaTime);
```

Comment:
Smoothing like this creates frame rate dependant results, which can be especially noticable if the game ever hitches and smoothFactor*deltaTime exceeds 1.
Can't remember where I saw it but I believe this gives a more consistent result:

```
float t = maxSmoothing * Game::instance.lastDeltaTime;
lerp(b, a, exp(-t));
```

Also since this is a getter function it shouldn't be updating the smoothed position as the code may want to call this function multiple times in one frame

Reply:
Thanks for the feedback ill have a play around with the smoothing, what do you mean by shouldn't update smoothed position, like I suppose if it could be called many times a frame it could result in slightly inconsistent smoothing but if I don't update it how will the lerp know where to be smoothing from to the new position? Apologies if this is a dumb question lol

Reply:
I have resolved some of these comments let me know what you think

Reply:
Sorry, you obviously have to update the smoothing somewhere or else, what I meant was you shouldn't being doing that inside GetDominantHandTransform. I would probably refactor the code to calculate the smoothed value once a frame (e.g. after UpdateInputs is called) and have the getter function return the current smoothed value.

Reply:
Other than that this is looking good.

Reply:
Should now be resolved